### PR TITLE
fix: Fix Deactivations API

### LINKED ISSUE #655: The "Deactivated Phone Numbers" API call is returning a 307 response causing an "Unable to fetch record" exception to be thrown
The example PHP code for Deactivated Phone Numbers displayed on https://www.twilio.com/docs/sms/deactivated-phone-numbers returns a 307 status code, which causes an Exception to be thrown by this code in the fetch method in src\Twilio\Version.php:

```PHP
if ($response->getStatusCode() < 200 || $response->getStatusCode() >= 300) {
    throw $this->exception($response, 'Unable to fetch record');
}
```

### PR DESCRIPTION
**Pull request by ssigwart**
# Fixes #

Fixes "Unable to fetch record" for Deactivated Phone Numbers API
Closes #655

### Checklist
- [X] I acknowledge that all my contributions will be made under the project's license
- [X] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [X] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [X] I have titled the PR appropriately
- [X] I have updated my branch with the main branch
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation about the functionality in the appropriate .md file
- [X] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by thinkingserious** [comment]
Internal tracking: DI-916

**Comment by childish-sambino** [comment]
This change should not be specific to certain endpoints. Changes should be made similar to what's laid out here: https://github.com/twilio/twilio-csharp/issues/534#issuecomment-676508898

**Comment by ssigwart** [comment]
Thanks, @childish-sambino.  I updated it based on that comment.

File: tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function testFetchRequest(): void {
 
     public function testFetchResponse(): void {
         $this->holodeck->mock(new Response(
-            200,
+            307,
```
Comment (childish-sambino) [inline tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php:35]: This is a generated file (see top of file) which will be overwritten on the next release.

File: src/Twilio/Version.php [hunk 2]
```diff
@@ -109,7 +109,7 @@ public function fetch(string $method, string $uri,
             $timeout
         );
 
-        if ($response->getStatusCode() < 200 || $response->getStatusCode() >= 300) {
+        if ($response->getStatusCode() < 200 || $response->getStatusCode() >= 400) {
```
Comment (childish-sambino) [inline src/Twilio/Version.php:112]: Were you able to verify locally that this fixes the issue you're seeing?

Comment (childish-sambino) [reply src/Twilio/Version.php:112]: Also, could you add a comment about why 3XX is allowed _just_ here and not in the other wrappers?

Comment (jdavidbakr) [reply src/Twilio/Version.php:112]: I can verify that this code change resolves the issue for me if I change the file locally.  To @childish-sambino's comment, this seems to be a unique endpoint response in that it is a redirect, I don't believe any of the other endpoints are redirects which is why 3XX is not expected there.

Comment (childish-sambino) [reply src/Twilio/Version.php:112]: Yup, just meant a comment in the code itself.

Comment (ssigwart) [reply src/Twilio/Version.php:113]: I added a comment.  I think that's what you wanted.

File: tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php (hunk 1 above)
Comment (ssigwart) [reply tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php:35]: Thanks.  I didn't see that.  If there some other place I should update?

Comment (childish-sambino) [reply tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php:35]: Check `VersionTest.php`

Comment (ssigwart) [reply tests/Twilio/Integration/Messaging/V1/DeactivationsTest.php]: Thanks.  I thought I was looking for something to test the deactivations API specifically.  I think I have it updated as you'd like now.

**Review by childish-sambino — APPROVED** [review]
LGTM

@ssigwart Can you merge from `main` so I can the merge this into `main`?

**Comment by ssigwart** [comment]
@childish-sambino, I ran the following:

```bash
git fetch upstream main
git merge upstream/main
git push origin deactivations-307
```

**Pull request merged**